sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/api/Microsoft.Azure.WebJobs.Extensions.EventHubs.netstandard2.0.cs

```diff
@@ -20,6 +20,15 @@ public sealed partial class EventHubTriggerAttribute : System.Attribute
 }
 namespace Microsoft.Azure.WebJobs.EventHubs
 {
+    public partial class EventHubAsyncCollector : Microsoft.Azure.WebJobs.IAsyncCollector<Azure.Messaging.EventHubs.EventData>, System.IDisposable
```

Comment:
Who is responsible for disposing this?

Reply:
The same party that was responsible before we made it public.    We didn't alter the contract, just the visibility for the concrete type versus just the interface projection.   My assumption is that it's part of the infrastructure's DI management.

Reply:
You are probably right, but I'm not certain that this wasn't an existing bug. My other concern is that users may think they need to dispose this type, which I assume would break their app.

Reply:
Would a concurrent dictionary allow us to avoid making this Disposable?

Reply:
I don't believe so.   The batch needs to stay unchanged between multiple stages in order to not lose events.   Under the semaphore, the following takes place:

- Create if not exists
- Attempt to add event to it
- If at capacity, pop it for send
- Create a new batch
- Add the current item

That said, it looks like the last two steps are not done, and the `item` that was added when the batch moves across the threshold gets dropped on the floor.   I need to poke at that to be sure that I'm not mistaken - but I think there's a bug that's been here.

Reply:
Nope, no bug.   I missed the `while(true)` - I added a test to validate as well.

Reply:
@JoshLove-msft :  Would it help to implement `IDisposable` explicitly so that `Dispose` doesn't appear for most cases?

Reply:
That is probably a good idea.

Reply:
Moving the binding to an extension method off the interface makes this moot as the type is internal again.